design-proposals/klip-4-produce-data.md

```diff
@@ -0,0 +1,277 @@
+> |                         | PARTITIONS and REPLICAS must be specified to create the topic. Command will fail if the topic exists |
+> |                         | with different partition/replica counts.                                                             |
+> +-------------------------+------------------------------------------------------------------------------------------------------+
+> | PARTITIONS              | The number of partitions in the backing topic. This property must be set if creating a SOURCE without|
+> |                         | an existing topic.                                                                                   |
+> +-------------------------+------------------------------------------------------------------------------------------------------+
+> | REPLICAS                | The number of replicas in the backing topic. This property must be set if creating a SOURCE without  |
+> |                         | an existing topic.                                                                                   |
+> +-------------------------+------------------------------------------------------------------------------------------------------+
+> ```
+* A new section for `INSERT INTO ... VALUES` will be added to the `sytnax.rst`:
+> ```rst
+> **Synopsis**
+> 
+> .. code:: sql
+>
+>   INSERT INTO stream_name/table_name [( column_name [, ...] )]
+>     VALUES (value [, ...])
+> 
+> **Description**
+> 
+> Produce a row into an existing stream or table and its underlying topic based on
+> explicitly sepcified values. The first ``column_name`` of every schema is ``ROWKEY``, which
+> defines the corresponding kafka key - it is expected to always match a corresponding column in 
+> the value if the create statement specifies a key.
+>
+> Column values can be given in several ways:
+> * Any column not explicitly given a value is set to ``null`` (KSQL does not support DEFAULT values)
+> * If no columns are specified, a value for every column is expected in the same order as the
+> schema
+>
+> The values will serialize using the ``value_format`` specified in the original `CREATE` statment.
+> ```
+* A new section for `DELETE` will be added to the `syntax.rst`:
+> ```rst
+> **Synopsis**
+> 
+> .. code:: sql
+> 
+>     DELETE FROM table_name WHERE condition;
+>     
+> **Description**
+> 
+> Delete a row from an existing table. The ``condition`` must be of form ``ROWKEY = value`` and the
+> ``table_name`` must represent a Table (as opposed to Stream). This will issue a tombstone, producing
+> a value to the underlying kafka topic with the specified key and a null value. Deleting a value that
+> does not exist will not affect the table contents.
+> ```
+
+
+# Compatibility Implications
+
+* All previous commands that succeeded will still succeed after this KLIP
+* `CREATE` commands that failed with "topic does not exist" may now fail with "must specify 
+partitions and replicas to create source with empty topic."
```

Comment:
as above, we should just make these a configurable default

Reply:
Disagree - see comment above.

Reply:
> Another thing to note... if the user provides PARTITIONS=3 and the topic already exists, but has 30 partitions... what does KSQL do? Fail? Continue?

My intention was that it would fail if anything about the topic is different. I will make this clear in the docs.